src/wallet/index.ts

```diff
@@ -181,12 +224,100 @@ class Wallet {
+   *  and verify that it matches the transaction prior to signing. This gives the user a sanity check
+   *  to ensure that what they try to encode matches the message that will be recieved by rippled.
+   *
+   * @param serialized - A signed and serialized transaction.
+   * @param tx - The transaction prior to signing.
+   * @throws A ValidationError if the transaction does not have a TxnSignature/Signers property, or if
+   * the serialized Transaction desn't match the original transaction.
+   */
+  // eslint-disable-next-line class-methods-use-this -- Helper for organization purposes
+  private checkTxSerialization(serialized: string, tx: Transaction): void {
+    // Decode the serialized transaction:
+    const decoded = decode(serialized)
+    const txCopy = { ...tx }
+
+    // ...And ensure it is equal to the original tx, except:
+    // - It must have a TxnSignature or Signers (multisign).
+    if (!decoded.TxnSignature && !decoded.Signers) {
+      throw new ValidationError(
+        'Serialized transaction must have a TxnSignature or Signers property',
+      )
+    }
+    // - We know that the original tx did not have TxnSignature, so we should delete it:
+    delete decoded.TxnSignature
+    // - We know that the original tx did not have Signers, so if it exists, we should delete it:
+    delete decoded.Signers
+
+    // - If SigningPubKey was not in the original tx, then we should delete it.
+    //   But if it was in the original tx, then we should ensure that it has not been changed.
+    if (!tx.SigningPubKey) {
+      delete decoded.SigningPubKey
+    }
+
+    // - Memos have exclusively hex data which should ignore case.
+    //   Since decode goes to upper case, we set all tx memos to be uppercase for the comparison.
+    txCopy.Memos?.map((memo) => {
+      const memoCopy = { ...memo }
+      if (memo.Memo.MemoData) {
+        memoCopy.Memo.MemoData = memo.Memo.MemoData.toUpperCase()
+      }
+
+      if (memo.Memo.MemoType) {
+        memoCopy.Memo.MemoType = memo.Memo.MemoType.toUpperCase()
+      }
+
+      if (memo.Memo.MemoFormat) {
+        memoCopy.Memo.MemoFormat = memo.Memo.MemoFormat.toUpperCase()
+      }
+
+      return memo
+    })
+    if (!_.isEqual(decoded, tx)) {
+      const data = {
+        decoded,
+        tx,
+      }
+      const error = new ValidationError(
+        'Serialized transaction does not match original txJSON. See error.data',
+        data,
+      )
+      throw error
+    }
+  }
+
+  /**
+   * Signs a transaction with the proper signing encoding.
+   *
+   * @param tx - A transaction to sign.
+   * @param privateKey - = A key to sign the transaction with.
+   * @param signAs - Multisign only. An account address to include in the Signer field.
+   * Can be either a classic address or an XAddress.
+   * @returns A signed transaction in the proper format.
+   */
+  // eslint-disable-next-line class-methods-use-this -- Helper to add organizational clarity
```

Comment:
This does actually need to take `this` as a param, because it uses `this.privateKey`

Reply:
Is it cleaner to accept 'this' explicitly as a parameter instead of just using it implicitly?